src/Misc/layoutroot/safe_sleep.sh

```diff
@@ -1,6 +1,16 @@
 #!/bin/bash
 
+if [ -x "$(command -v sleep)" ]; then
+  sleep $1
+  exit 0
+fi
+if [ -x "$(command -v ping)" ]; then
+  # Add 1 since the first ping is executed immediately.
+  ping -c $(( $1 + 1 )) 127.0.0.1 > /dev/null
+  exit 0
+fi
+
 SECONDS=0
-while [[ $SECONDS != $1 ]]; do
+while [[ $SECONDS -lt $1 ]]; do
```

Comment:
[`+` and `-lt` are vulnerable to code-injection](https://mywiki.wooledge.org/BashProgramming/05#Arithmetic_Expansion) ([SC](https://github.com/koalaman/shellcheck/issues/3067))

Previously [mentioned here](https://github.com/actions/runner/pull/3157#issuecomment-3638892152)